src/main/kotlin/no/nav/familie/ba/sak/kjerne/simulering/KontrollerNyUtbetalingsgeneratorService.kt

```diff
@@ -0,0 +1,193 @@
+package no.nav.familie.ba.sak.kjerne.simulering
+
+import no.nav.familie.ba.sak.common.secureLogger
+import no.nav.familie.ba.sak.config.FeatureToggleService
+import no.nav.familie.ba.sak.integrasjoner.økonomi.AndelTilkjentYtelseForSimuleringFactory
+import no.nav.familie.ba.sak.integrasjoner.økonomi.UtbetalingsoppdragGeneratorService
+import no.nav.familie.ba.sak.integrasjoner.økonomi.ØkonomiKlient
+import no.nav.familie.ba.sak.kjerne.behandling.domene.Behandling
+import no.nav.familie.ba.sak.kjerne.simulering.domene.SimuleringsPeriode
+import no.nav.familie.ba.sak.kjerne.simulering.domene.tilTidslinje
+import no.nav.familie.ba.sak.kjerne.tidslinje.Tidslinje
+import no.nav.familie.ba.sak.kjerne.tidslinje.fraOgMed
+import no.nav.familie.ba.sak.kjerne.tidslinje.komposisjon.kombinerMed
+import no.nav.familie.ba.sak.kjerne.tidslinje.tidspunkt.Måned
+import no.nav.familie.ba.sak.kjerne.tidslinje.tidspunkt.tilForrigeMåned
+import no.nav.familie.ba.sak.kjerne.tidslinje.tilOgMed
+import no.nav.familie.ba.sak.kjerne.tidslinje.transformasjon.beskjær
+import no.nav.familie.ba.sak.kjerne.vedtak.Vedtak
+import no.nav.familie.ba.sak.sikkerhet.SikkerhetContext
+import no.nav.familie.kontrakter.felles.oppdrag.Utbetalingsoppdrag
+import no.nav.familie.kontrakter.felles.simulering.DetaljertSimuleringResultat
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.math.BigDecimal
+
+@Service
+class KontrollerNyUtbetalingsgeneratorService(
+    private val featureToggleService: FeatureToggleService,
+    private val økonomiKlient: ØkonomiKlient,
+    private val utbetalingsoppdragGeneratorService: UtbetalingsoppdragGeneratorService,
+) {
+
+    private val logger = LoggerFactory.getLogger(KontrollerNyUtbetalingsgeneratorService::class.java)
```

Comment:
Tips er å bruke `javaClass` direkte, så trenger man ikke hele klassenavnet :-) 
```suggestion
    private val logger = LoggerFactory.getLogger(javaClass)
```